gcloud-java-dns/src/main/java/com/google/gcloud/dns/Dns.java

```diff
@@ -420,5 +420,191 @@ public static ChangeRequestListOption sortOrder(SortingOrder order) {
+   * end of the list. Use {@link ZoneListOption} to restrict the listing to a domain name, set page
+   * size, and set page tokens.
+   *
+   * @return {@code Page<Zone>}, a page of zones
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/managedZones/list">Cloud DNS Managed Zones:
+   * list</a>
+   */
+  Page<Zone> listZones(ZoneListOption... options);
+
+  /**
+   * Deletes an existing zone identified by name. Returns true if the zone was successfully deleted
+   * and false otherwise.
+   *
+   * @return {@code true} if zone was found and deleted and false otherwise
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/managedZones/delete">Cloud DNS Managed Zones:
+   * delete</a>
+   */
+  boolean delete(String zoneName); // delete does not admit any options
+
+  /**
+   * Deletes an existing zone identified by id. Returns true if the zone was successfully deleted
+   * and false otherwise.
+   *
+   * @return {@code true} if zone was found and deleted and false otherwise
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/managedZones/delete">Cloud DNS Managed Zones:
+   * delete</a>
+   */
+  boolean delete(BigInteger zoneId); // delete does not admit any options
+
+  /**
+   * Lists the DNS records in the zone identified by name.
+   *
+   * <p>The fields to be returned, page size and page tokens can be specified using {@code
+   * DnsRecordOptions}. Returns null if the zone cannot be found.
+   *
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/resourceRecordSets/list">Cloud DNS
+   * ResourceRecordSets: list</a>
+   */
+  Page<DnsRecord> listDnsRecords(String zoneName, DnsRecordListOption... options);
+
+  /**
+   * Lists the DNS records in the zone identified by ID.
+   *
+   * <p>The fields to be returned, page size and page tokens can be specified using {@code
+   * DnsRecordOptions}. Returns null if the zone cannot be found.
+   *
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/resourceRecordSets/list">Cloud DNS
+   * ResourceRecordSets: list</a>
+   */
+  Page<DnsRecord> listDnsRecords(BigInteger zoneId, DnsRecordListOption... options);
+
+  /**
+   * Retrieves the metadata about the current project. The returned fields can be optionally
+   * restricted by specifying {@code ProjectOptions}.
+   *
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/projects/get">Cloud DNS Projects: get</a>
+   */
+  ProjectInfo getProjectInfo(ProjectGetOption... fields);
+
+  /**
+   * Returns the current project id.
+   */
+  String getProjectId();
+
+  /**
+   * Returns the current project number.
+   */
+  BigInteger getProjectNumber();
+
+  /**
+   * Submits a change requests for applying to the zone identified by ID to the service. The
```

Comment:
It might be me here but the meaning of "Submits a change requests for applying to the zone identified by ID to the service." is not clear to me.